wiki/Ranking_Criteria/osu!taiko/en.md

```diff
@@ -3,46 +3,161 @@ osu!taiko
 
 _Main page: [Ranking Criteria](/wiki/Ranking_Criteria)_
 
-These rules and guidelines has been discussed in the [osu!taiko Discussion Thread](https://osu.ppy.sh/forum/p/1275323). New rules or guidelines may be suggested in the [discussion forum](https://osu.ppy.sh/forum/87) which are going to be updated once the discussion reaches an agreement.
+These rules and guidelines have been discussed in the [Ranking Criteria subforum](https://osu.ppy.sh/community/forums/87). New rules or guidelines may be suggested in that forum and the criteria on this page will be updated once that discussion reaches an agreement.
 
-Common Terms
+Glossary
 --------------
 
-### General
+### Difficulty Naming
+- ![Kantan](/wiki/shared/Easy-t.png "Kantan") **Kantan:** Easy
+- ![Futsuu](/wiki/shared/Normal-t.png "Futsuu") **Futsuu:** Normal
+- ![Muzukashii](/wiki/shared/Hard-t.png "Muzukashii") **Muzukashii:** Hard
+- ![Oni](/wiki/shared/Insane-t.png "Oni") **Oni:** Insane
+- ![Inner/Ura Oni](/wiki/shared/Expert-t.png "Inner/Ura Oni") **Inner/Ura Oni:** Expert
 
--   **Kantan**: Easy
--   **Futsuu**: Normal
--   **Muzukashii**: Hard
--   **Oni**: Demon/Insane
--   **Inner** and **Ura**: Extra/Expert
+Refer to [this thread](https://osu.ppy.sh/community/forums/topics/178700) for alternative difficulty naming conventions.
 
-### Gameplay
+The prefix `Inner` or `Ura` may be used to indicate a higher difficulty tier than the one of its base but lower than the next one (e.g. Ura Kantan is harder than Kantan, but easier than Futsuu).
 
--   **Don**: Red Note
--   **Kat**: Blue Note
--   **Drumroll**: Yellow Bar
--   **Shaker**: Spinner
--   **Finisher**: Large Blue or Red Note
+### Gameplay elements
+- **Don / Red Note:** A beat that can be hit with any of the Drum Center attached keys (default keys `X`, `C`).
+- **Kat / Blue Note:** A beat that can be hit with any of the Drum Rim attached keys (default keys `Z`, `V`).
+- **Big Don / Big Red Note / Finisher Note:** A strong beat that can be hit with both Drum Center keys at the same time for extra score.
+- **Big Kat / Big Blue Note / Finisher Note:**  A strong beat that can be hit with both Drum Rim keys at the same time for extra score.
+- **BPM:** An acronym for “beats per minute” used to determine the tempo of a song.
+- **Slider / Drumroll:** A yellow bar with ticks inside of it which are played with any of the assigned keys. These ticks are usually snapped to 1/4 corresponding to the song's BPM. If the BPM is lower than 125 this changes to 1/8 snapping and 1/2 snapping if the BPM is higher than 250. If slider tick rate is set to 3, the slider ticks will instead snap to 1/3.
```

Comment:
replace "If the BPM is lower than 125 this changes to 1/8 snapping and 1/2 snapping if the BPM is higher than 250." with "If the BPM is lower than 125, the snapping will change to 1/8. If the BPM is higher than 250, it will change to 1/2." for consistency with the rest of the article